pkg/skaffold/build/tag/tag_template_test.go

```diff
@@ -18,10 +18,98 @@ package tag
 
 import (
 	"testing"
+	"time"
 
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 	"github.com/GoogleContainerTools/skaffold/testutil"
 )
 
+func TestTagTemplate_GenerateTag(t *testing.T) {
+	aLocalTimeStamp := time.Date(2015, 03, 07, 11, 06, 39, 123456789, time.Local)
+
+	dateTimeExample := &dateTimeTagger{
+		Format:   "2006-01-02",
+		TimeZone: "UTC",
+		timeFn:   func() time.Time { return aLocalTimeStamp },
+	}
+
+	envTemplateExample, _ := NewEnvTemplateTagger("{{.FOO}}")
+	invalidEnvTemplate, _ := NewEnvTemplateTagger("{{.BAR}}")
+	env := []string{"FOO=BAR"}
+
+	tagTemplateExample, _ := NewTagTemplateTagger("", nil)
```

Comment:
Suggest also check the return error for NewTagTemplateTagger since this is a new function. (see the test coverage regression).

Reply:
The first test case "empty template" tests `NewTagTemplateTagger("", nil)` before `tagTemplateExample` is used and checks that there is no error. Is this sufficient?

Reply:
I suppose this is fine to ignore the error since you're just creating an example. could you also add a separate test for this function though? something like `TestNewTagger()` that calls `NewTagTemplateTagger()` with different sets of params and checks error/success

Reply:
Sounds good to me.